File: Assignment2/src/parser2/DfaUnionNode.java
package parser2;

public class DfaUnionNode {
	public DfaNode node1;
	public DfaNode node2;
	
	public DfaUnionNode On0 = null;
	public DfaUnionNode On1 = null;
	public DfaUnionNode On2 = null;
	
	public boolean IsStart() {
		return node1.IsStart && node2.IsStart;
	}
	
	public boolean IsFinal() {
		return node1.IsFinal && node2.IsFinal;
	}
	
	public boolean inspected = false;

}

File: Assignment2/src/logic/DfaUnionNode.scala
package logic

class DfaUnionNode {
  var node1: DfaNode = null
  var node2: DfaNode = null

  var On0: DfaUnionNode = null
  var On1: DfaUnionNode = null
  var On2: DfaUnionNode = null

  def IsStart(): Boolean = {
    return node1.IsStart && node2.IsStart;
  }

  def IsFinal(): Boolean = {
    return node1.IsFinal && node2.IsFinal;
  }

  var inspected = false;
}